From d538db7a3b689beb1242b5c2b774ce0d42478805 Mon Sep 17 00:00:00 2001
From: Corey Bryant <corey.bryant@canonical.com>
Date: Tue, 05 May 2020 15:42:31 -0400
Subject: [PATCH] Monkey patch original current_thread _active

Monkey patch the original current_thread to use the up-to-date _active
global variable. This solution is based on that documented at:
https://github.com/eventlet/eventlet/issues/592

Change-Id: I61e5e270bf66b0355da3282c19cbc9fd42c4090b
Closes-Bug: #1863021
---

diff --git a/trove/cmd/__init__.py b/trove/cmd/__init__.py
index c579ba3..617f6f1 100644
--- a/trove/cmd/__init__.py
+++ b/trove/cmd/__init__.py
@@ -26,3 +26,9 @@
 if not os.environ.get('NO_EVENTLET_MONKEYPATCH'):
     import eventlet
     eventlet.monkey_patch(all=True)
+    # Monkey patch the original current_thread to use the up-to-date _active
+    # global variable. See https://bugs.launchpad.net/bugs/1863021 and
+    # https://github.com/eventlet/eventlet/issues/592
+    import __original_module_threading as orig_threading
+    import threading  # noqa
+    orig_threading.current_thread.__globals__['_active'] = threading._active
